src/Provisioning/Provisioning.csproj

```diff
@@ -0,0 +1,251 @@
+	</Target>
+
+	<Target Name="ProvisionAndroidSdkAvdCreateAvds">
+
+		<!-- Create all the AVD's we need based on the api levels specified -->
+		<Exec Command="dotnet android avd create --name &quot;Emulator_%(AndroidSdkApiLevels.Identity)&quot; --sdk &quot;system-images%3Bandroid-%(AndroidSdkApiLevels.Identity)%3B%(AndroidSdkApiLevels.SystemImageType)%3B$(AndroidSdkHostAbi)&quot; --device &quot;%(AndroidSdkApiLevels.DeviceType)&quot; --force" />
+	</Target>
+
+	<Target Name="ProvisionAndroidSdk" DependsOnTargets="ProvisionAndroidSdkCommonPackages;ProvisionAndroidSdkPlatformApiPackages;ProvisionAndroidSdkEmulatorImagePackages;ProvisionAndroidSdkAvdCreateAvds">
+		<Error
+			Condition=" '$(MSBuildRuntimeType)' != 'Core' "
+			Text="You should only run this target with 'dotnet build'." />
+	</Target>
+
+
+	<!-- ******************************************** -->
+	<!-- * Appium Provisioning                        -->
+	<!-- ******************************************** -->
+
+	<Target Name="ProvisionAppiumDrivers" DependsOnTargets="InstallAppium">
+
+		<!-- Get a json output list of the installed appium drivers with their version info -->
+		<Exec Command="appium driver list --installed --json" ConsoleToMSBuild="True" IgnoreExitCode="True">
+			<Output TaskParameter="ConsoleOutput" ItemName="_AppiumDriverListInstalledJsonLines" />
+		</Exec>
+
+		<!-- Concatenate the item group console output lines into one single property string -->
+		<PropertyGroup>
+			<_AppiumDriverListInstalledJson>@(_AppiumDriverListInstalledJsonLines->'%(Identity)', ' ')</_AppiumDriverListInstalledJson>
+		</PropertyGroup>
+
+		<!-- Get the version of each driver from the JSON output and put it in a property -->
+		<JsonPeek Content="$(_AppiumDriverListInstalledJson)" Query="$.%(AppiumDrivers.Identity).version">
+			<Output TaskParameter="Result" PropertyName="_AppiumVersion_%(AppiumDrivers.Identity)" />
+		</JsonPeek>
+
+		<!-- Get the groups of drivers to install vs update based on the versions resolved and expected -->
+		<ItemGroup>
+			<AppiumDriversToInstall Include="@(AppiumDrivers)" KeepMetadata="Version" Condition=" '$(_AppiumVersion_%(AppiumDrivers.Identity))' == '' " />
+			<AppiumDriversToUpdate Include="@(AppiumDrivers)" KeepMetadata="Version" Condition=" '$(_AppiumVersion_%(AppiumDrivers.Identity))' != '' AND '$(_AppiumVersion_%(AppiumDrivers.Identity))' != '%(AppiumDrivers.Version)' " />
+		</ItemGroup>
+
+
+		<!-- Update any drivers needing an update (if any) -->
+		<Exec Command="appium driver update %(AppiumDriversToUpdate.Identity)" Condition=" '@(AppiumDriversToUpdate)' != '' " />
+
+		<!-- Install any drivers not yet installed (if any) -->
+		<Exec Command="appium driver install %(AppiumDriversToInstall.Identity)@%(AppiumDriversToInstall.Version)" Condition=" '@(AppiumDriversToInstall)' != '' " />
+
+		<!-- Run doctor command just so we have some output -->
```

Comment:
Can we add a Condition to the doctor , it caused issues like we saw on the agents where on macOS it hanged checking something with Xcode, and we can have that flag off to macOS  for now like I did in my last commit to main